### ISSUE TITLE
マルチタスク学習の実装

### ISSUE THREAD

**Issue by synx-kawai** [synx-kawai/wavira#57]
## 概要
WhoFi（人物識別）との統合学習や補助タスクを追加し、特徴表現を強化する。

## 背景
単一タスク（混雑レベル分類）だけでなく、関連タスクを同時に学習することで、より汎用的で頑健な特徴表現を獲得できる可能性がある。

## タスク
- [ ] WhoFiとの統合学習
  - [ ] 共有エンコーダ + タスク別ヘッド構造の設計
  - [ ] 混雑推定 + 人物識別の同時学習
  - [ ] 損失関数の重み付け調整
- [ ] 補助タスクの追加
  - [ ] RSSI予測（回帰タスク）
  - [ ] 動き検出（2値分類：動きあり/なし）
  - [ ] ノイズレベル推定
- [ ] マルチタスク学習フレームワーク
  - [ ] 動的タスク重み調整（Uncertainty Weighting等）
  - [ ] Gradient Normalization
- [ ] カリキュラム学習
  - [ ] 簡単なタスクから難しいタスクへ段階的に学習

## 実装場所
`wavira/models/multitask.py` (新規作成)
`wavira/training/multitask_trainer.py` (新規作成)

## 評価
- シングルタスク vs マルチタスクの精度比較
- 補助タスクの有効性検証

## 成果物
- [ ] マルチタスク学習モジュール
- [ ] 統合トレーニングスクリプト
- [ ] 比較実験結果

## 関連
- Phase 3: モデルアーキテクチャの改善

**Comment by synx-kawai** [comment]
## 完了報告

### 実装内容

#### 新規作成ファイル
- `wavira/models/multitask.py` - マルチタスク学習モデル
- `wavira/training/multitask_trainer.py` - マルチタスクトレーナー
- `tests/test_multitask.py` - テストコード（26テスト）

### 実装機能

#### マルチタスク学習モデル
- **MultiTaskCrowdEstimator**: 共有エンコーダ + タスク別ヘッド構造
- 主タスク: 混雑レベル分類
- 補助タスク:
  - RSSI予測（回帰）
  - 動き検出（2値分類）
  - ノイズレベル推定（回帰）

#### タスク重み付け
- **Uncertainty Weighting**: 学習可能なタスク重み（Kendall et al., 2018）
- **GradientNormalizer**: 勾配正規化による動的タスクバランス

#### トレーニング
- **MultiTaskTrainer**: 完全なトレーニングループ
- **カリキュラム学習**: 補助タスクの段階的導入
- チェックポイント保存/読み込み

### 使用方法

```python
from wavira.models.multitask import create_multitask_model
from wavira.training.multitask_trainer import train_multitask_model

# モデル作成
model = create_multitask_model(enable_auxiliary_tasks=True)

# トレーニング
model, history = train_multitask_model(
    train_loader,
    val_loader,
    n_epochs=50,
    use_curriculum=True,
)
```

### 動作確認
- 全26テストパス
